### PR TITLE
Match HTTP_HOST in pair with relative URL

### PR DESCRIPTION
**Pull request by uzzer**
For compatibility matching with relative URL
only left as it was before.

### CODE REVIEW

File: lib/gaffe/errors_controller_resolver.rb [hunk 1]
```diff
@@ -30,7 +30,11 @@ def resolved_controller
   private
 
     def request_controller(controllers)
-      controllers.find { |pattern, _| @env['REQUEST_URI'] =~ pattern }.try(:last)
+      controllers.find do |pattern, _|
+        relative_url = @env['REQUEST_URI']
+        absolute_url = @env['HTTP_HOST'] + relative_url
+        !(relative_url =~ pattern).nil? || !(absolute_url =~ pattern).nil?
+      end.try(:last)
```
Comment (remiprev) [inline lib/gaffe/errors_controller_resolver.rb]: I like the idea but could you fix these two things:

1. The condition is very difficult to read/understand at first glance. Can you rewrite it to be more verbose?
2. I don’t like `end.<method>`. The result of `#find` whould be stored in a variable and we should call `try(:last)` on it.

File: spec/gaffe/errors_controller_resolver_spec.rb [hunk 2]
```diff
@@ -42,6 +43,16 @@
         it { expect(controller).to eql :api_controller }
       end
 
+      context 'with error coming from matching HTTP_HOST' do
+        let(:env) do
+          request.env.merge(
+            'HTTP_HOST' => 'www.com',
+            'REQUEST_URI' => '/'
+          )
+        end
```
Comment (remiprev) [inline spec/gaffe/errors_controller_resolver_spec.rb]: Even if we merge two keys in the `env`, I think we should keep the `let` on a single line, like with the other contexts.

```ruby
let(:env) { request.env.merge 'HTTP_HOST' => 'www.com', 'REQUEST_URI' => '/' }
```

**Comment by remiprev** [comment]
I think we should adapt the `README.md` to mention that we support this.

**Comment by uzzer** [comment]
Changes applied according to comments. 

Me personal opinion is that matching system currently quite amateur and should not be over-engineered in the beginning.

**Comment by uzzer** [comment]
Can I anyhow help? 💤

**Comment by remiprev** [comment]
@uzzer Can you rebase your commits into a single one?

**Comment by uzzer** [comment]
@remiprev done

**Comment by remiprev** [comment]
Alright, thanks!

**Pull request merged**